pkg/analytics/analytics.go

```diff
@@ -71,46 +128,33 @@ func New(ctx context.Context, opts ...Option) (*LogRecorder, error) {
 		sdklog.WithProcessor(sdklog.NewBatchProcessor(stdoutExporter)),
 	)
 
-	return &LogRecorder{
-		provider: loggerProvider,
-	}, nil
+	global.SetLoggerProvider(loggerProvider)
+	return nil
+}
+
+type shutdownLoggerProvider interface {
+	otellog.LoggerProvider
+	Shutdown(ctx context.Context) error
 }
 
-func (a *LogRecorder) Stop(ctx context.Context) error {
-	defer func() {
-		if err := a.provider.Shutdown(ctx); err != nil {
-			log.Warn().Err(err).Msg("failed to shutdown analytics")
+func ShutdownAnalyticsProvider(ctx context.Context) error {
+	provider, ok := global.GetLoggerProvider().(*sdklog.LoggerProvider)
+	if ok {
+		if err := provider.ForceFlush(ctx); err != nil {
+			log.Trace().Err(err).Msg("failed to flush analytics log provider")
```

Comment:
You are still force flushing. Does it mean it won't delay shutdown?

Reply:
`provider.ForceFlush` should respect context cancellation of the context provided to it, but I'll just remove this call.